modules/cnf-image-based-upgrade-prep.adoc

```diff
@@ -0,0 +1,120 @@
+    namespace: openshift-lifecycle-agent
+  oadpContent: <6>
+  - name: oadp-cm-example
+    namespace: openshift-adp
+----
+<1> Specify the target platform version. The value must match the version of the seed image.
+<2> Specify the repository where the target cluster can pull the seed image from.
+<3> Specify the reference to a secret with credentials to pull container images if the images are in a private registry.
+<4> (Optional) Specify the time frame in seconds to roll back if the upgrade does not complete within that time frame after the first reboot. If not defined or set to `0`, the default value of `1800` seconds (30 minutes) is used.
+<5> (Optional) Specify the list of `ConfigMap` resources that contain your custom catalog sources to retain after the upgrade, and your extra manifests to apply to the target cluster that are not part of the seed image.
+<6> Add the `oadpContent` section with the OADP `ConfigMap` information.
+
+. When you are ready to start the `Prep` stage, change the value of the `stage` field to `Prep` in the `ImageBasedUpgrade` CR:
++
+--
+[source,terminal]
+----
+$ oc patch imagebasedupgrades.lca.openshift.io upgrade -p='{"spec": {"stage": "Prep"}}' --type=merge -n openshift-lifecycle-agent
+----
+
+If you provide `ConfigMap` objects for OADP resources and extra manifests,, {lcao} validates the specified `ConfigMap` objects during the `Prep` stage.
+You may encounter the following issues: 
+
+* validation warnings or errors if the {lcao} detects any issues with `extraManifests` 
+* validation errors if the {lcao} detects any issues with with `oadpContent`
+
+Validation warnings do not block the `Upgrade` stage but you must decide if it is safe to proceed with the upgrade.
+These warnings, for example missing CRDs, namespaces, or dry run failures, update the `status.conditions` for the `Prep` stage and `annotation` fields in the `ImageBasedUpgrade` CR with details about the warning.
+
+.Example validation warning
+[source,yaml]
+----
+[...]
+metadata:
+annotations:
+  extra-manifest.lca.openshift.io/validation-warning: '...'
+[...]
+----
+
+However, validation errors, such as adding `MachineConfig` or Operator manifests to extra manifests, cause the `Prep` stage to fail and block the `Upgrade` stage.
+
+When the validations pass, the cluster creates a new `ostree` stateroot, which involves pulling and unpacking the seed image, and running host level commands.
+Finally, all the required images are precached on the target cluster.
+--
+
+.Verification
+
+. Check the status of the `ImageBasedUpgrade` CR.
++
+--
+[source,terminal]
+----
+$ oc get ibu -A -oyaml
+----
+
+.Example output
+[source,yaml]
+----
+status:
+  conditions:
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: In progress
+    observedGeneration: 2
+    reason: InProgress
+    status: "False"
+    type: Idle
+  - lastTransitionTime: "2024-01-01T09:00:00Z"
+    message: 'Prep completed: total: 121 (pulled: 1, skipped: 120, failed: 0)'
```

Comment:
The example output should be updated for upgrade stage as well, see "Upgrade completed:" from https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/image-based-upgrade.md#starting-the-upgrade-stage